docs/Menu-Commands.md

```diff
@@ -0,0 +1,78 @@
+
+## Contents
+
+- [Defining a new command](#defining-a-new-command)
+- [Contributing context menu items](#contributing-context-menu-items)
+  - [Zowe Explorer context menu group naming convention](#zowe-explorer-context-menu-group-naming-convention)
+
+## Defining a new commmand
+
+In order to add a command to a core Zowe Explorer menu, the command must first be defined in `packages/zowe-explorer/package.json`, under [`contributes.commands`](https://code.visualstudio.com/api/references/contribution-points#contributes.commands). The following conventions should be used when defining the respective command properties:
+
+- `command` property: If the command is specific to one of Zowe Explorer's views (i.e. Data Sets, USS, or Jobs view), one of the prefixes in the table below should be added at the front of this property's value to indicate what Zowe Explorer view the command applies to.
+
+  | Prefix                | Applicable view | Example usage           |
+  | --------------------- | --------------- | ----------------------- |
+  | `zowe.<command>`      | Data Sets       | `zowe.addFavorite`      |
+  | `zowe.uss.<command>`  | USS             | `zowe.uss.addFavorite`  |
+  | `zowe.jobs.<command>` | Jobs            | `zowe.jobs.addFavorite` |
+
+- `title` property: Be sure to localize the command title's value. This can be done by following the steps below:
+
+  1. Create a concise, descriptive key for the title string and wrap it in `%` on each side. Consider prefixing the key name with `mvs.`, `uss.`, or `jobs.` to indicate if the key is for the Data Sets, USS, or Jobs view, respectively. (For example, the key `%uss.addFavorite%` is used for the `addFavorite` command in the USS view.) If the command is more related to profiles than a specific view, consider prefixing the key name with `profile.`.
+  1. in `package.nls.json`, define the user-facing string for the key. (For example: `"uss.addFavorite": "Add to Favorites"`.)
+
+  - More details on localization can be found in the [Developer's ReadMe](https://github.com/zowe/vscode-extension-for-zowe/blob/master/docs/Developer's%20ReadMe.md#adding-strings).
+
+- `category` property: The value of this should be "Zowe Explorer".
+
+**Note:** By default, defined commands will appear in VS Code's Command Palette. If the command being added is not designed to work from the Command Palette, be sure to add an entry for the command in the `contributes.menus.commandPalette` section of `packages/zowe-explorer/package.json`, and specify `"when": "never"`. (For more details, see VS Code's documentation on [specifying command visibility in the Command Palette](https://code.visualstudio.com/api/references/contribution-points#Context-specific-visibility-of-Command-Palette-menu-items).)
```

Comment:
@katelynienaber This note is for the Zowe Explorer squad devs, because it seems we have a tendency to forget to do this, and it previously resulted in many extra commands in the Command Palette that I had to remove with PR #1308. I feel it's important (and forgotten) enough that we want to mention it when defining the command. I didn't have any additional guidance I wanted to add for the Command Palette (and thus didn't make a section for it), but if there is guidance you think should be added, we can create a section for it.

Reply:
I rephrased this section to begin with the action item: "If the command is not designed to work from the Command Palette, be sure to...". However, I also added a bit more text to explain the rationale for following this guidance.